numpy/core/src/common/simd/neon/math.h

```diff
@@ -0,0 +1,60 @@
+#ifndef NPY_SIMD
+    #error "Not a standalone header"
+#endif
+
+#ifndef _NPY_SIMD_NEON_MATH_H
+#define _NPY_SIMD_NEON_MATH_H
+
+/***************************
+ * Elementary
+ ***************************/
+// Absolute
+#define npyv_abs_f32 vabsq_f32
+#define npyv_abs_f64 vabsq_f64
+
+// Square
+NPY_FINLINE npyv_f32 npyv_square_f32(npyv_f32 a)
+{ return vmulq_f32(a, a); }
+#if NPY_SIMD_F64
+    NPY_FINLINE npyv_f64 npyv_square_f64(npyv_f64 a)
+    { return vmulq_f64(a, a); }
+#endif
+
+// Square root
+#if NPY_SIMD_F64
+    #define npyv_sqrt_f32 vsqrtq_f32
+    #define npyv_sqrt_f64 vsqrtq_f64
+#else
+    // Based on ARM doc, see https://developer.arm.com/documentation/dui0204/j/CIHDIACI
+    NPY_FINLINE npyv_f32 npyv_sqrt_f32(npyv_f32 a)
+    {
+        const npyv_f32 zero = vdupq_n_f32(0.0f);
+        // estimate to (1/√a)
+        npyv_f32 rsqrte = vrsqrteq_f32(a);
+        // fine-tune it with Newton-Raphson and vrsqrtsq_f32((3.0 + (a * -square)) / 2.0)
+        rsqrte = vmulq_f32(rsqrte, vrsqrtsq_f32(a, vmulq_f32(rsqrte, rsqrte)));
+        rsqrte = vmulq_f32(rsqrte, vrsqrtsq_f32(a, vmulq_f32(rsqrte, rsqrte)));
+        // a * (1/√a)
+        npyv_f32 sqrt = vmulq_f32(a, rsqrte);
+        // return zero if the a is zero
+        npyv_u32 bits = vbicq_u32(vreinterpretq_u32_f32(sqrt), vceqq_f32(a, zero));
```

Comment:
The OpenCV simd sqrt is [here](https://github.com/opencv/opencv/blob/master/modules/core/include/opencv2/core/hal/intrin_neon.hpp#L887). but you are right, if we didn't check zero, the result will become a nan.

Reply:
thanks, to #16782, I catched two issues here positive infinity(same as zero) and precision.
unfortunately, I had to add a third Newton-Raphson iteration to provides acceptable precision.

Reply:
Can you add a test for that failure?

Reply:
I made a check to all sqrt special cases, see:
https://github.com/numpy/numpy/blob/1f872658984b2f8b0fda7022e72ad333a62864f3/numpy/core/tests/test_simd.py#L184-L188

I made another fix to `npyv_sqrt_f32()` within the last push, which fixes floating-point division-by-zero error that
raised by `vrsqrteq_f32(x)` when x is zero.

Now all the tests passed on armhf.